ndb/src/google/cloud/ndb/eventloop.py

```diff
@@ -30,16 +31,228 @@
 ]
 
 
-def add_idle(*args, **kwargs):
-    raise NotImplementedError
+def _noop(*args, **kw):
+    """Do nothing."""
+
+# TODO: Use utils.logging_debug when implemented
+_logging_debug = _noop
+
+
+_Event = collections.namedtuple('_Event', (
+    'when', 'callback', 'args', 'kwargs'))
+
+
+class _Clock(object):
```

Comment:
This is from the original source and is basically just a site for dependency injection for testing.  Could use `unittest.mock.patch` instead.  Might be more consistent with modern usage.

Reply:
:+1:  for using `mock.patch('time.time')` or `mock.patch('time.sleep')` where needed.

Reply:
9512caff2315b79c35b0abd4fdd4db6449466c1d